src/convert.jl

```diff
@@ -248,15 +248,40 @@ Converts a hyperrectangular set to a zonotope.
 
 ### Input
 
-- `Zonotope`
-- `H` -- hyperrectangular set
+- `Zonotope` -- type, used for dispatch
+- `H`        -- hyperrectangular set
 
 ### Output
 
 A zonotope.
 """
-function convert(::Type{Zonotope}, H::AbstractHyperrectangle)
-    return Zonotope(center(H), Diagonal(radius_hyperrectangle(H)))
+function convert(::Type{Zonotope}, H::AbstractHyperrectangle{N}) where {N}
```

Comment:
the implementation was wrong; i added a new one. not sure though if we should keep this `if isflat(H)` branch, since, if it's not flat, the code in lines 260-276 is also ok.